From 62100b84d7a302b9806aaffbacea4df57fd12ce7 Mon Sep 17 00:00:00 2001
From: Justin Hammond <justin@dynam.ac>
Date: Fri, 5 May 2023 01:37:56 +0800
Subject: [PATCH] fix include path

---
 Accounts/AccountsQt5Config.cmake.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Accounts/AccountsQt5Config.cmake.in b/Accounts/AccountsQt5Config.cmake.in
index 55a8964..047e2c1 100644
--- a/Accounts/AccountsQt5Config.cmake.in
+++ b/Accounts/AccountsQt5Config.cmake.in
@@ -1,5 +1,7 @@
 #  ACCOUNTSQT_INCLUDE_DIRS - The libaccounts-qt include directories
 #  ACCOUNTSQT_LIBRARIES - The libraries needed to use libaccounts-qt
 
+get_filename_component(PACKAGE_PREFIX_DIR ${CMAKE_CURRENT_LIST_DIR}/../../../ ABSOLUTE)
+
 set(ACCOUNTSQT_LIBRARIES $${INSTALL_LIBDIR}/lib$${TARGET}.so)
-set(ACCOUNTSQT_INCLUDE_DIRS $${INSTALL_PREFIX}/include/$${TARGET}/)
\ No newline at end of file
+set(ACCOUNTSQT_INCLUDE_DIRS ${PACKAGE_PREFIX_DIR}/include/$${TARGET}/)
\ No newline at end of file
